process/smartContract/builtInFunctions/factory.go

```diff
@@ -93,3 +112,33 @@ func CreateBuiltInFunctionsFactory(args ArgsCreateBuiltInFunctionContainer) (vmc
 
 	return bContainerFactory, nil
 }
+
+func GetAllowedAddress(coordinator sharding.Coordinator, addresses [][]byte) ([]byte, error) {
+	if check.IfNil(coordinator) {
+		return nil, process.ErrNilShardCoordinator
+	}
+
+	if len(addresses) == 0 {
+		return nil, fmt.Errorf("%w for shard %d, provided count is %d", process.ErrNilCrawlerAllowedAddress, coordinator.SelfId(), len(addresses))
+	}
+
+	sortedAddresses := make([][]byte, len(addresses))
+	copy(sortedAddresses, addresses)
+
+	sort.Slice(sortedAddresses, func(i, j int) bool {
```

Comment:
we do not really need this sort. but it can stay.

Reply:
yup, we do not need this sort

Reply:
removed

Reply:
removed